Add JSON format output

JamesSmaldon opened the pull request:
Added JSON output to lsdvd, it can be enabled using the -Oj flag.

JamesSmaldon committed:
Add JSON format output

thierer commented:
Thanks for the PR! I thought about adding JSON support myself but never got around doing it...

Just a minor nitpick: Is there a specific reason why you didn't remove the [LFs for return_hash_outer, return_hash_inner and return_array_inner in the ruby syntax](https://github.com/thierer/lsdvd/blob/e955ae1291cb691b6854181f6180d757a25634c8/lsdvd-0.16/ocode.c#L100-L103) like you did with the other formats? 

I don't know a thing about ruby and never used this output, it's just that it now looks a bit odd with the extra LFs.

JamesSmaldon committed:
Improve formatting for Ruby code output

JamesSmaldon committed:
Fix printf related compiler warnings.

JamesSmaldon committed:
Add JSON output option to command help and man page.

JamesSmaldon commented:
No problem, I forgot about the ruby LFs but I've removed them now and slightly tweaked the output and it looks better. I also fixed a few warnings while I was in the area and updated the command help and manpage to include the new option.

thierer commented:
Merged, thanks again! 

(I rearranged your commits a bit to separate the formatting fixes from the functional changes).
